### PR TITLE
[SandboxVec][SeedCollector] Reject non-simple memory ops for memory seeds

### PR DESCRIPTION
**Pull request by Sterling-Augustine**
Load/Store isSimple is a necessary condition for VectorSeeds, but not sufficient, so reverse the condition and return value, and continue the check. Add relevant tests.

### CODE REVIEW

**Comment by vporpo** [comment]
Please add a `[SandboxVec][SeedCollector]` or similar prefix to the title.

File: llvm/unittests/Transforms/Vectorize/SandboxVectorizer/SeedCollectorTest.cpp [hunk 1]
```diff
@@ -466,5 +472,48 @@ define void @foo(ptr noalias %ptr, float %v, <2 x float> %val) {
...
+
+  sandboxir::Context Ctx(C);
+  auto &F = *Ctx.createFunction(&LLVMF);
+  auto BB = F.begin();
+  sandboxir::SeedCollector SC(&*BB, SE);
+
+  // Find the stores
+  auto It = std::next(BB->begin(), 2);
+  // StX with X as the order by offset in memory
+  auto *Ld0 = &*It++;
```
Comment (vporpo) [inline llvm/unittests/Transforms/Vectorize/SandboxVectorizer/SeedCollectorTest.cpp:510]: I would change this to `auto *Ld0 = cast<sandboxir::LoadInst>(&*It++);` just to make sure that the iterator is actually pointing to a load.

File: llvm/unittests/Transforms/Vectorize/SandboxVectorizer/SeedCollectorTest.cpp [hunk 2]
```diff
@@ -466,5 +472,48 @@ define void @foo(ptr noalias %ptr, float %v, <2 x float> %val) {
...
+  auto *Ld1 = &*It++;
```
Comment (vporpo) [inline llvm/unittests/Transforms/Vectorize/SandboxVectorizer/SeedCollectorTest.cpp:511]: same here.

File: llvm/lib/Transforms/Vectorize/SandboxVectorizer/SeedCollector.cpp [hunk 3]
```diff
@@ -140,8 +140,8 @@ LLVM_DUMP_METHOD void SeedContainer::dump() const { print(dbgs()); }
 #endif // NDEBUG
 
 template <typename LoadOrStoreT> static bool isValidMemSeed(LoadOrStoreT *LSI) {
-  if (LSI->isSimple())
-    return true;
+  if (!LSI->isSimple())
```
Comment (slackito) [inline llvm/lib/Transforms/Vectorize/SandboxVectorizer/SeedCollector.cpp:143]: Is there any test for cases where the LoadOrStoreT is simple, but one of the later checks fails?

**Review by vporpo — APPROVED** [review]
A couple of comments, looks good otherwise.

File: llvm/lib/Transforms/Vectorize/SandboxVectorizer/SeedCollector.cpp (hunk 3 above)
Comment (Sterling-Augustine) [reply llvm/lib/Transforms/Vectorize/SandboxVectorizer/SeedCollector.cpp:143]: added

File: llvm/unittests/Transforms/Vectorize/SandboxVectorizer/SeedCollectorTest.cpp (hunk 1 above)
Comment (Sterling-Augustine) [reply llvm/unittests/Transforms/Vectorize/SandboxVectorizer/SeedCollectorTest.cpp]: done

File: llvm/unittests/Transforms/Vectorize/SandboxVectorizer/SeedCollectorTest.cpp (hunk 2 above)
Comment (Sterling-Augustine) [reply llvm/unittests/Transforms/Vectorize/SandboxVectorizer/SeedCollectorTest.cpp]: done

**Review by slackito — APPROVED** [review]

**Pull request merged**